docs/content/en/docs/reference/clusterspec/cloudstack.md

```diff
@@ -5,4 +5,346 @@ weight: 30
+A list of labels to apply to the nodes in the worker node group. This is in addition to the labels that
+EKS Anywhere will add by default.
+For example, this label identifies a Cluster API failuredomain, which causes CAPC to substitute the value for the actual AZ name as a label on the nodes:
+
+```yaml
+    labels:
+      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
+```
+
+Modifying the labels associated with a worker node group configuration will cause new nodes to be rolled out, replacing
+the existing nodes associated with the configuration.
+
+## CloudStackDatacenterConfig
+
+### availabilityZones.account (optional)
+Account used to access CloudStack.
+The default is `admin`.
+As long as you pass valid credentials, through `availabilityZones.credentialsRef`, this value is not required.
+
+### availabilityZones.credentialsRef (optional)
+If you passed credentials through the environment variable `EKSA_CLOUDSTACK_B64ENCODED_SECRET` noted in [Create CloudStack production cluster]({{< relref "../../getting-started/production-environment/cloudstack.md" >}}), you can identify those credentials here.
+For that example, you would use the value `global`.
+You can also use a previously created Kubernetes secret.
+
+### availabilityZones.domain (required)
+CloudStack domain to deploy the cluster. The default is `ROOT`.
+
+### availabilityZones.managementApiEndpoint (required)
+Location of the CloudStack API management endpoint. For example, `http://10.11.0.2:8080/client/api`.
+
+### availabilityZones.name (required)
+Name of the CloudStack zone on which to deploy the cluster.
+As an alternative, you could pass a Zone ID to `availabilityZones.id`.
+
+### availabilityZones.zone.network.name (required)
+CloudStack network name to use with the cluster.
+
+## CloudStackMachineConfig
+In the example above, there are separate `CloudStackMachineConfig` sections for the control plane (`my-cluster-name-cp`), worker (`my-cluster-name`) and etcd (`my-cluster-name-etcd`) nodes.
+
+### computeOfferings (required)
+Name or ID of the CloudStack compute instance.
+
+### users[0].name (optional)
+The name of the user you want to configure to access your virtual machines through ssh.
+
+The default is `capc`.
+
+### users[0].sshAuthorizedKeys (optional)
+The SSH public keys you want to configure to access your virtual machines through ssh (as described below). Only 1 is supported at this time.
+
+### users[0].sshAuthorizedKeys[0] (optional)
+This is the SSH public key that will be placed in `authorized_keys` on all EKS Anywhere cluster VMs so you can ssh into
+them. The user will be what is defined under name above. For example:
+
+```
+ssh -i <private-key-file> <user>@<VM-IP>
+```
+
+The default is generating a key in your `$(pwd)/<cluster-name>` folder when not specifying a value.
+
+### template (required)
+The VM template to use for your EKS Anywhere cluster. Currently, a VM based on RHEL 8.6 is required.
+This can be a name or ID.
```

Comment:
to be clear, it would look like
```
template:
  name: xxxx
```

or 

```
template:
  id: yyyy
```

so one of template.name or template.id are required. Same for a lot of the other fields that reference CloudStack resources

Reply:
I used the {id,name} notation on each field it applies to. I'll go through later and see if I missed any.